core/client/src/main/java/alluxio/client/block/BlockStoreContext.java

```diff
@@ -237,8 +203,9 @@ public void releaseWorkerClient(BlockWorkerClient blockWorkerClient) {
     // close the client.
     if (blockWorkerClient.isLocal()) {
       // Return local worker client to its resource pool.
-      Preconditions.checkState(mLocalBlockWorkerClientPool != null);
-      mLocalBlockWorkerClientPool.release(blockWorkerClient);
+      WorkerNetAddress address = blockWorkerClient.getWorkerNetAddress();
+      Preconditions.checkState(mLocalBlockWorkerClientPoolMap.containsKey(address));
+      mLocalBlockWorkerClientPoolMap.get(address).release(blockWorkerClient);
```

Comment:
Although we use a concurrent data structure here, this logic is still not thread safe right?